pkg/ottl/ottlfuncs/func_replace_pattern.go

```diff
@@ -57,7 +57,7 @@ func applyReplaceFormat[K any](ctx context.Context, tCtx K, replacementFormat ot
 			return "", errFmt
 		}
 		if !validFormatString(formatStringVal) {
-			return "", fmt.Errorf("replacementFormat must be format string containing a single %%s and no other format specifiers")
+			return "", fmt.Errorf("replacementFormat must be format string containing a single %%s and no other format specifiers") //nolint:revive
```

Comment:
Is there a reason we can't use `errors.New` here? The `%%s` is just a normal string in this case

Reply:
Ah, got it. Is it too much trouble to change the string to use `%s` with `errors.New`?

Reply:
I am not familiar with the `ottlfuncs` pkg so ignore the comment if it doesn't make sense - the nolint just caught my eye.